### PR TITLE
GeoTiff Doc for MasterDocs

### PR DESCRIPTION
**Pull request by jbouffard**
This PR adds an updated overview of GeoTiffs to the master docs.

### CODE REVIEW

File: docs/raster/Geotiffs.md [hunk 1]
```diff
@@ -3,51 +3,55 @@
...
       - [Striped](#striped)
       - [Tiled](#tiled)
     - [Layout of Columns and Rows](#layout-of-columns-and-rows)
+- [Big Tiffs](#big-tiffs)
 - [Further Readings](#further-reading)
-- [To Do](#to-do)
 
 - - -
 ### Introduction
-GeoTiffs are a type of Tiff image file that contain image data pertaining to satiellete, arial, and evelation imagery amoung other types of geospatial information. What differentiates them from convential Tiffs is that additional metadata pertaining to the image is embedded wihtin the file. The positions of geographic features on the screen and how they are projected are two such examples of information found within GeoTiffs but lacking in a normal Tiff.
```
Comment (fosskers) [inline docs/raster/Geotiffs.md]: amoung -> among

File: docs/raster/Geotiffs.md [hunk 2]
```diff
@@ -3,51 +3,55 @@
...
 - - -
+
 ### GeoTiff File Format
-Because GeoTiffs are Tiffs wtih extended features, they have both have the same file structure. There exists three components that can be found in Tiff files: the header, file directory, and the actual image data. Within these files, the directories and image data can be found at any point within the file, regardless of how the images are presented when the file is opended and viewed. The header is the only section which has a constant location and that is at the begining of the file.
+Because GeoTiffs are Tiffs with extended features, they both have the same file structure. There exists three components that can be found in all Tiff files: the header, the image file directory, and the actual image data. Within these files, the directories and image data can be found at any point within the file; regardless of how the images are presented when the file is opended and viewed. The header is the only section which has a constant location, and that is at the begining of the file.
```
Comment (fosskers) [inline docs/raster/Geotiffs.md]: There exists -> There exist

File: docs/raster/Geotiffs.md [hunk 3]
```diff
@@ -3,51 +3,55 @@
...
 
 #### File Header
-As stated earlier, the header is found at the beginning of every Tiff file including GeoTiffs. Due to how the other components can be situated anywhere within the file, the header provides the offset value that points to the first file directory. Without this offset, it would be impossible to read a Tiff file.
+As stated earlier, the header is found at the beginning of every Tiff file, including GeoTiffs. All Tiff files have the exact same header size of 8 bytes. The first two bytes of the header are used to determine the `ByteOrder` of the file, also known as "Endianness". After these two, comes the next two bytes which are used to determine the file's magic number. `.tif`, `.txt`, `.shp`, and all other file types have a unique identifier number that tells the program kind of file it was give. For Tiff files, the magic number is 42. Due to how the other components can be situated anywhere within the file, the  last 4 bytes of the header provide the offset value that points to the first file directory. Without this offset, it would be impossible to read a Tiff file.
```
Comment (fosskers) [inline docs/raster/Geotiffs.md]: it was give -> it was given

File: docs/raster/Geotiffs.md [hunk 4]
```diff
@@ -3,51 +3,55 @@
...
-For each piece of image data found in a Tiff there exists a corresponding file directory for that image. The directory itself holds details about the image. Each property listed in the directory is reffered to as a, "Tag". Tags contain information on, but not limited to, the image size, compression types, and the type of color plan. Since we're working with GeoTiffs, important geographical information is also included such as the coordinate system the image is in and the latitudal and longitudal extent of the image. These directories can vary in size, as users can create their own tags and each image in the file does not need to have exact same tags.
+For every image found in a Tiff file there exists a corresponding image file directory for that picture. Each property listed in the directory is reffered to as a, `Tag`. `Tag`s contain information on, but not limited to, the image size, compression types, and the type of color plan. Since we're working with Geotiffs, geo-spatial information is also documented within the `Tag`s. These directories can vary in size, as users can create their own tags and each image in the file does not need to have exact same tags.
```
Comment (fosskers) [inline docs/raster/Geotiffs.md]: reffered -> referred

Comment (fosskers) [inline docs/raster/Geotiffs.md]: as a, `Tag` -> as a `Tag`

File: docs/raster/Geotiffs.md [hunk 5]
```diff
@@ -3,51 +3,55 @@
...
 
 Other than image attributes, the file directory holds two offset values that play a role in reading the file. One points to where the actual image itself is located, and the other shows where the the next file directory can be found.
 
 #### Image Data
-A Tiff file can store any number of images wihtin a single file, including none at all. In the case of GeoTiffs, the images themselves are almost always stored as bitmap data. It is important to understand that there are two ways in which the image can be stored. These two storage methods are: Striped and Tiled.
+
+A Tiff file can store any number of images wihtin a single file, including none at all. In the case of GeoTiffs, the images themselves are almost always stored as bitmap data. It is important to understand that there are two ways in which the actual image data is formated within the file. the two methods are: Striped and Tiled.
```
Comment (fosskers) [inline docs/raster/Geotiffs.md]: wihtin -> within

Comment (fosskers) [inline docs/raster/Geotiffs.md]: formated -> formatted

Comment (fosskers) [inline docs/raster/Geotiffs.md]: the two methods -> The two methods

File: docs/raster/Geotiffs.md [hunk 6]
```diff
@@ -3,51 +3,55 @@
...
 
 ##### Striped
-As seen in the picture below, the stiped storage format splits each image into long, vertical strips. Contained within them are columns of bitmapped image data. If your GeoTiff file was created before the realse of Tiff 6.0, then this is the data storage method in which it most likely uses.
+Striped storage breaks the image into segments of long, vertical bands that stretch the entire width of the pictue. Contained within them are columns of bitmapped image data. If your GeoTiff file was created before the realse of Tiff 6.0, then this is the data storage method in which it most likely uses.
```
Comment (fosskers) [inline docs/raster/Geotiffs.md]: pictue -> picture

Comment (fosskers) [inline docs/raster/Geotiffs.md]: realse -> release

File: docs/raster/Geotiffs.md [hunk 7]
```diff
@@ -3,51 +3,55 @@
...
 
-If an image has strip storage, then it's corresponding file directory contains the tags: RowsPerStrip, StripOffsets, and StripByteCount. All three of these are needed to read the image. The first one is the number of rows that are contained within the strips. Every strip wihtin an image must have the same number of rows within it except for the last one in certain instances. StripOffsets is an array of offsets that shows where each strip starts. The last tag, ByteSegmentCount, is also an array of values that contains the size of each strip in terms of Bytes.
+If an image has strip storage, then it's corresponding file directory contains the tags: `RowsPerStrip`, `StripOffsets`, and `StripByteCount`. All three of these are needed to read that given segment. The first one is the number of rows that are contained within the strips. Every strip wihtin an image must have the same number of rows within it except for the last one in certain instances. `StripOffsets` is an array of offsets that shows where each strip starts within the file. The last tag, `ByteSegmentCount`, is also an array of values that contains the size of each strip in terms of Bytes.
```
Comment (fosskers) [inline docs/raster/Geotiffs.md]: then it's corresponding -> then its corresponding

File: docs/raster/Geotiffs.md [hunk 8]
```diff
@@ -3,51 +3,55 @@
...
 
-As with stips, tiles have specific tags that are needed in order to read the image. These new tags are: TileWidth, TileLength, TileOffsets, and TileByteCounts. TileWidth is the number of coloumns and TileLength is the number of rows that are found within the specified tile. As with striped, TileOffsets and TileByteCounts are arrays that contain the begining offset and the byte of each tile in the image respectivly.
+As with stips, tiles have specific tags that are needed in order to process each segment. These new tags are: `TileWidth`, `TileLength`, `TileOffsets`, and `TileByteCounts`. `TileWidth` is the number of coloumns and `TileLength` is the number of rows that are found within the specified tile. As with striped, `TileOffsets` and `TileByteCounts` are arrays that contain the begining offset and the byte count of each tile in the image, respectivly.
```
Comment (fosskers) [inline docs/raster/Geotiffs.md]: respectivly -> respectively

File: docs/raster/Geotiffs.md [hunk 9]
```diff
@@ -3,51 +3,55 @@
...
 
 #### Layout of Columns and Rows
-There exists two ways in which to describe a location in GeoTiffs. One is in Map coordinates which use X and Y values. X's are oriented along the horizontal axis and run from west to east while Y's are on the vertical axis and run from south to north.
+There exists two ways in which to describe a location in GeoTiffs. One is in Map coordinates which use X and Y values. X's are oriented along the horizontal axis and run from west to east while Y's are on the vertical axis and run from south to north. Thus the further east you are, the larger your X value ; and the more north you are the larger your Y value.
```
Comment (fosskers) [inline docs/raster/Geotiffs.md]: There exists -> There exist

File: docs/raster/Geotiffs.md [hunk 10]
```diff
@@ -3,51 +3,55 @@
...
+The other method is to use the grid coordinate system. This technique of measurement uses Cols and Rows to describe the relative location of things. Cols run east to west whereas Rows run north to south. This then means that Cols increase as you go east to west, and rows increase as you go north to south.
 - - -
-### Further Readings
-* [More info on the Tiff file format](http://www.fileformat.info/format/tiff/egff.htm)
-* [More info on the GeoTiff file format](http://www.gdal.org/frmt_gtiff.html)
 
+### Big Tiffs
+In some instances, your GeoTiff may contain an amount of data so large that it can no longer be descriped as a Tiff, but rather by a new name, BigTiff. In order to qualify as a BigTiff, your file needs to be **at least 4gb in size or larger**. At this point, the methods used to store and find data need to be changed. The accomodation that is made is to change the size of the various offsets and byte counts of each segment. For a normal Tiff, this size is 32-bits, but BigTiffs have these sizes at 64-bit. GeoTrellis supports BigTiffs without any issue, so one doesn not need to worry about size when working with their files.
```
Comment (fosskers) [inline docs/raster/Geotiffs.md]: so one doesn not need to worry -> so one need not worry

**Comment by fosskers** [comment]
:beer:

**Comment by jbouffard** [comment]
:beers:

File: docs/raster/Geotiffs.md [hunk 11]
```diff
@@ -3,51 +3,122 @@
...
       - [Striped](#striped)
       - [Tiled](#tiled)
     - [Layout of Columns and Rows](#layout-of-columns-and-rows)
+- [Big Tiffs](#big-tiffs)
 - [Further Readings](#further-reading)
-- [To Do](#to-do)
 
 - - -
 ### Introduction
-GeoTiffs are a type of Tiff image file that contain image data pertaining to satiellete, arial, and evelation imagery amoung other types of geospatial information. What differentiates them from convential Tiffs is that additional metadata pertaining to the image is embedded wihtin the file. The positions of geographic features on the screen and how they are projected are two such examples of information found within GeoTiffs but lacking in a normal Tiff.
```
Comment (rossbernet) [inline docs/raster/Geotiffs.md]: satiellete -> satellite

File: docs/raster/Geotiffs.md [hunk 12]
```diff
@@ -3,51 +3,122 @@
...
+GeoTiffs are a type of Tiff image file that contain image data pertaining to
+satiellete, arial, and evelation data among other types of geospatial
+information. The additional pieces of metadata that are needed to store and
+display this information is what sets GeoTiffs apart from norml Tiffs. For
```
Comment (rossbernet) [inline docs/raster/Geotiffs.md]: norml > normal

File: docs/raster/Geotiffs.md [hunk 13]
```diff
@@ -3,51 +3,122 @@
...
+absent from a normal Tiff file.
 - - -
+
 ### GeoTiff File Format
-Because GeoTiffs are Tiffs wtih extended features, they have both have the same file structure. There exists three components that can be found in Tiff files: the header, file directory, and the actual image data. Within these files, the directories and image data can be found at any point within the file, regardless of how the images are presented when the file is opended and viewed. The header is the only section which has a constant location and that is at the begining of the file.
+Because GeoTiffs are Tiffs with extended features, they both have the same file
+structure. There exist three components that can be found in all Tiff files:
+the header, the image file directory, and the actual image data. Within these
+files, the directories and image data can be found at any point within the file;
+regardless of how the images are presented when the file is opended and viewed.
```
Comment (rossbernet) [inline docs/raster/Geotiffs.md]: opended > opened

File: docs/raster/Geotiffs.md [hunk 14]
```diff
@@ -3,51 +3,122 @@
...
+Other than image attributes, the file directory holds two offset values that
+play a role in reading the file. One points to where the actual image itself is
+located, and the other shows where the the next file directory can be found.
 
-Other than image attributes, the file directory holds two offset values that play a role in reading the file. One points to where the actual image itself is located, and the other shows where the the next file directory can be found.
 
 #### Image Data
-A Tiff file can store any number of images wihtin a single file, including none at all. In the case of GeoTiffs, the images themselves are almost always stored as bitmap data. It is important to understand that there are two ways in which the image can be stored. These two storage methods are: Striped and Tiled.
+A Tiff file can store any number of images wihtin a single file, including none
```
Comment (rossbernet) [inline docs/raster/Geotiffs.md]: wihtin > within

File: docs/raster/Geotiffs.md [hunk 15]
```diff
@@ -3,51 +3,122 @@
...
+Striped storage breaks the image into segments of long, vertical bands that
+stretch the entire width of the picture. Contained within them are columns of
+bitmapped image data. If your GeoTiff file was created before the realse of Tiff
+6.0, then this is the data storage method in which it most likely uses.
 
-If an image has strip storage, then it's corresponding file directory contains the tags: RowsPerStrip, StripOffsets, and StripByteCount. All three of these are needed to read the image. The first one is the number of rows that are contained within the strips. Every strip wihtin an image must have the same number of rows within it except for the last one in certain instances. StripOffsets is an array of offsets that shows where each strip starts. The last tag, ByteSegmentCount, is also an array of values that contains the size of each strip in terms of Bytes.
+If an image has strip storage, then its corresponding file directory contains
+the tags: `RowsPerStrip`, `StripOffsets`, and `StripByteCount`. All three of
+these are needed to read that given segment. The first one is the number of rows
+that are contained within the strips. Every strip wihtin an image must have the
```
Comment (rossbernet) [inline docs/raster/Geotiffs.md]: wihtin > within

File: docs/raster/Geotiffs.md [hunk 16]
```diff
@@ -3,51 +3,122 @@
...
+same number of rows within it except for the last one in certain instances.
+`StripOffsets` is an array of offsets that shows where each strip starts within
+the file. The last tag, `ByteSegmentCount`, is also an array of values that
+contains the size of each strip in terms of Bytes.
 
 ##### Tiled
-Tiff 6.0 introduced tiled storage for image data. These rectangluar segments have both a height and a width that must be divisible by 16. There are instances where the tiled grid does not fit the image exactly. When this occurs, padding is added around the image so as to meet the requirement of having each tile having dimensions of a factor of 16. See the image below for a visual representation of what this format looks like.
+Tiff 6.0 introduced a new way to arrange and store data within a Tiff, tiled
+storage. These rectangluar segments have both a height and a width that must be
```
Comment (rossbernet) [inline docs/raster/Geotiffs.md]: rectangluar > rectangular

File: docs/raster/Geotiffs.md [hunk 17]
```diff
@@ -3,51 +3,122 @@
...
+divisible by 16. There are instances where the tiled grid does not fit the image
+exactly. When this occurs, padding is added around the image so as to meet the
+requirement of each tile having dimensions of a factor of 16.
 
-As with stips, tiles have specific tags that are needed in order to read the image. These new tags are: TileWidth, TileLength, TileOffsets, and TileByteCounts. TileWidth is the number of coloumns and TileLength is the number of rows that are found within the specified tile. As with striped, TileOffsets and TileByteCounts are arrays that contain the begining offset and the byte of each tile in the image respectivly.
+As with stips, tiles have specific tags that are needed in order to process each
+segment. These new tags are: `TileWidth`, `TileLength`, `TileOffsets`, and
+`TileByteCounts`. `TileWidth` is the number of coloumns and `TileLength` is the
```
Comment (rossbernet) [inline docs/raster/Geotiffs.md]: coloumns > columns

File: docs/raster/Geotiffs.md [hunk 18]
```diff
@@ -3,51 +3,122 @@
...
+run east to west whereas Rows run north to south. This then means that Cols
+increase as you go east to west, and rows increase as you go north to south.
 - - -
-### Further Readings
-* [More info on the Tiff file format](http://www.fileformat.info/format/tiff/egff.htm)
-* [More info on the GeoTiff file format](http://www.gdal.org/frmt_gtiff.html)
 
+### Big Tiffs
+In some instances, your GeoTiff may contain an amount of data so large that it
+can no longer be descriped as a Tiff, but rather by a new name, BigTiff. In
```
Comment (rossbernet) [inline docs/raster/Geotiffs.md]: described

File: docs/raster/Geotiffs.md [hunk 19]
```diff
@@ -3,51 +3,122 @@
...
+order to qualify as a BigTiff, your file needs to be **at least 4gb in size or
+larger**. At this point, the methods used to store and find data need to be
+changed. The accomodation that is made is to change the size of the various
```
Comment (rossbernet) [inline docs/raster/Geotiffs.md]: accommodation

File: docs/raster/Geotiffs.md [hunk 20]
```diff
@@ -3,51 +3,122 @@
...
+offsets and byte counts of each segment. For a normal Tiff, this size is
+32-bits, but BigTiffs have these sizes at 64-bit. GeoTrellis supports BigTiffs
+without any issue, so one need not worry about size when working with
+their files.
 - - -
-### To Do
-Added the pictures of how the image data is layed out.
+
+### Further Readings
+* [For more information on the Tiff file format](http://www.fileformat.info/format/tiff/egff.htm)
```
Comment (rossbernet) [inline docs/raster/Geotiffs.md]: information

File: docs/raster/Geotiffs.md [hunk 21]
```diff
@@ -3,51 +3,122 @@
...
+* [For more informatioin on the GeoTiff file format](http://www.gdal.org/frmt_gtiff.html)
```
Comment (rossbernet) [inline docs/raster/Geotiffs.md]: information

File: docs/raster/Geotiffs.md (hunk 20 above)
Comment (fosskers) [inline docs/raster/Geotiffs.md]: Looks fine to me?

File: docs/raster/Geotiffs.md [hunk 22]
```diff
@@ -3,51 +3,122 @@
...
     - [Layout of Columns and Rows](#layout-of-columns-and-rows)
+- [Big Tiffs](#big-tiffs)
 - [Further Readings](#further-reading)
-- [To Do](#to-do)
 
 - - -
 ### Introduction
-GeoTiffs are a type of Tiff image file that contain image data pertaining to satiellete, arial, and evelation imagery amoung other types of geospatial information. What differentiates them from convential Tiffs is that additional metadata pertaining to the image is embedded wihtin the file. The positions of geographic features on the screen and how they are projected are two such examples of information found within GeoTiffs but lacking in a normal Tiff.
+GeoTiffs are a type of Tiff image file that contain image data pertaining to
+satellite, arial, and evelation data among other types of geospatial
```
Comment (fosskers) [inline docs/raster/Geotiffs.md]: Spelling here.

**Pull request merged**